### PR TITLE
Add support for thread data parsing.

### PR DESCRIPTION
**Pull request by emypar**
Provide access to /proc/PID/task/TID files, for thread specific values. Since such files have the same structure as /proc/PID/ ones, the data structures and the parsers for the latter may be reused.

@discordianfish, @pgier if you could take a look at this very basic PR, that would be appreciated. TIA

### CODE REVIEW

**Comment by emypar** [comment]
Hello @discordianfish, could you take a look at this simple PR? Thank you!

**Review by discordianfish — APPROVED** [review]
Great, LGTM!

**Comment by emypar** [comment]
> Great, LGTM!

Thank you, could you also merge it on my behalf or point me to ppl. that might be able to do so? I don't have write access.

**Comment by discordianfish** [comment]
We need another maintaoner to approve /cc @pgier @SuperQ

File: thread.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// Copyright 2018 The Prometheus Authors
```
Comment (SuperQ) [inline thread.go:1]:
```suggestion
// Copyright 2022 The Prometheus Authors
```

File: thread_test.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
+// Copyright 2018 The Prometheus Authors
```
Comment (SuperQ) [inline thread_test.go:1]:
```suggestion
// Copyright 2022 The Prometheus Authors
```

**Review by SuperQ — CHANGES_REQUESTED** [review]
Minor nit, otherwise LGTM.

File: thread.go (hunk 1 above)
Comment (emypar) [reply thread.go]: Done

File: thread_test.go (hunk 2 above)
Comment (emypar) [reply thread_test.go]: Done

**Comment by emypar** [comment]
> Minor nit, otherwise LGTM.

Thank you, all done!

**Review by SuperQ — APPROVED** [review]

**Comment by emypar** [comment]
Thank you all for making this happen.

**Pull request merged**